app/assets/javascripts/incurred_incidentals/form.js

```diff
@@ -0,0 +1,27 @@
+$(document).ready(function () {
+  $("#multiUploadMoreFields").click(function () {
+    if($(".multi-upload-group:visible").length < 1) {
+      $(".multi-upload-group").show(); //we never deleted it, just hid it
+    } else {
+      $(".multi-upload-group").first().clone().appendTo("#multiUploadContainer"); //add new group
+
+      //reset id's
+      var num = $(".multi-upload-group").length;
+      var group = $(".multi-upload-group").last(); //the grouping we just created
+
+      var fileInput = group.find('.multi-upload-file'); //find file input in the group
+      fileInput.attr("name",fileInput.attr("name").replace(/\d/,num)); //change the number in the id
+
+      var descInput = group.find('.multi-upload-desc'); //find file desc in the group
+      descInput.attr("name",descInput.attr("name").replace(/\d/,num)); //change the number in the id
+    }
+  });
+
+  $(".multiUploadDeleteField").click(function (e) {
+    if($(".multi-upload-group").length > 1) {
+      e.currentTarget.parentElement.remove();
```

Comment:
A little weird to be in jQuery, but to use the native Javascript DOM descriptors.

Should be `$(this).parent().remove();`.